src/controller/python/chip/icd/__init__.py

```diff
@@ -0,0 +1,100 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import asyncio
+from ctypes import CFUNCTYPE, c_uint8, c_uint64
+from dataclasses import dataclass
+from threading import Condition, Event, Lock
+from typing import Callable
+
+from chip import native
+
+_OnCheckInCompleteFunct = CFUNCTYPE(None, c_uint8, c_uint64)
+
+_OnCheckInCompleteWaitListLock = Lock()
+_OnCheckInCompleteWaitList = dict()
+
+
+@dataclass
```

Comment:
Please don't add complex logic and globals to `__init__.py`. Use `__init__.py` just to manage exports for `from xxx import *`. This logic can be put in a properly named file, preferably inside python/chip. I don't see a need to sub-package this under `python/chip/icd`.